### PR TITLE
Remove all --name and --root options

### PR DESCRIPTION
**Pull request by rhatdan**
This is a follow on to the remove all --link options

### CODE REVIEW

File: tests/basic.bats [hunk 1]
```diff
@@ -23,73 +23,73 @@ load helpers
...
-	newroot=$(buildah mount --name=$newcid)
+	newroot=$(buildah mount $newcid)
 	test -s $newroot/randomfile
 	cmp ${TESTDIR}/randomfile $newroot/randomfile
 	cp ${TESTDIR}/other-randomfile $newroot/other-randomfile
-	buildah commit --signature-policy ${TESTSDIR}/policy.json --name=$newcid --output=containers-storage:other-new-image
+	buildah commit --signature-policy ${TESTSDIR}/policy.json $newcid containers-storage:other-new-image
 	# Not an allowed ordering of arguments and flags.  Check that it's rejected.
-	run buildah commit --signature-policy ${TESTSDIR}/policy.json        $newcid --output=containers-storage:rejected-new-image
+	run buildah commit $newcid  --signature-policy ${TESTSDIR}/policy.json ontainers-storage:rejected-new-image
```
Comment (nalind) [inline tests/basic.bats]: The output image's transport name is typoed.

**Comment by nalind** [comment]
Multiple `ArgsUsage` field values in `main.go` also need updates.

File: tests/add.bats [hunk 2]
```diff
@@ -7,29 +7,29 @@ load helpers
...
+	buildah add $cid ${TESTDIR}/randomfile
 	# Copy a file to a specific subdirectory
-	buildah add --name=$cid --dest=/subdir ${TESTDIR}/randomfile
+	buildah add --dest=/subdir $cid ${TESTDIR}/randomfile
 	# Copy a file two files to a specific subdirectory
-	buildah add --name=$cid --dest=/other-subdir ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile
+	buildah add --dest=/other-subdir  $cid ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile
 	# Copy a file two files to a specific location, which fails because it's not a directory.
-	run buildah add --name=$cid --dest=/notthereyet-subdir ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile
+	run buildah add --dest=/notthereyet-subdir ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile  $cid
```
Comment (nalind) [inline tests/add.bats]: Stray whitespace at the end of this line?

File: tests/add.bats [hunk 3]
```diff
@@ -7,29 +7,29 @@ load helpers
...
 	# Copy a file to a different working directory
-	buildah config --workingdir=/cwd --name=$cid
-	buildah add --name=$cid ${TESTDIR}/randomfile
-	buildah unmount --name=$cid
-	buildah commit --signature-policy ${TESTSDIR}/policy.json --name=$cid --output=containers-storage:new-image
-	buildah delete --name=$cid
+	buildah config --workingdir=/cwd $cid
+	buildah add $cid ${TESTDIR}/randomfile
+	buildah unmount $cid
+	buildah commit --signature-policy ${TESTSDIR}/policy.json $cid containers-storage:new-image
```
Comment (nalind) [inline tests/add.bats]: Stray whitespace at the end of this line?

File: tests/basic.bats (hunk 1 above)
Comment (nalind) [inline tests/basic.bats]: Stray whitespace at the end of this line?

File: tests/copy.bats [hunk 4]
```diff
@@ -7,19 +7,19 @@ load helpers
...
 	cid=$(buildah from --pull --signature-policy ${TESTSDIR}/policy.json --image alpine)
-	root=$(buildah mount --name=$cid)
-	buildah config --name=$cid --workingdir /
-	buildah copy --name=$cid ${TESTDIR}/randomfile
-	buildah copy        $cid ${TESTDIR}/other-randomfile
-	buildah unmount --name=$cid
-	buildah commit --signature-policy ${TESTSDIR}/policy.json --name=$cid --output=containers-storage:new-image
-	buildah delete --name=$cid
+	root=$(buildah mount $cid)
+	buildah config --workingdir / $cid
```
Comment (nalind) [inline tests/copy.bats]: Stray whitespace at the end of this line?

File: tests/copy.bats [hunk 5]
```diff
@@ -7,19 +7,19 @@ load helpers
...
-	buildah copy        $cid ${TESTDIR}/other-randomfile
-	buildah unmount --name=$cid
-	buildah commit --signature-policy ${TESTSDIR}/policy.json --name=$cid --output=containers-storage:new-image
-	buildah delete --name=$cid
+	root=$(buildah mount $cid)
+	buildah config --workingdir / $cid 
+	buildah copy $cid ${TESTDIR}/randomfile
+	buildah copy $cid ${TESTDIR}/other-randomfile
+	buildah unmount $cid
+	buildah commit --signature-policy ${TESTSDIR}/policy.json $cid containers-storage:new-image
```
Comment (nalind) [inline tests/copy.bats]: Stray whitespace at the end of this line?

File: tests/add.bats (hunk 2 above)
Comment (rhatdan) [inline tests/add.bats]: fixed

File: tests/add.bats (hunk 3 above)
Comment (rhatdan) [inline tests/add.bats]: Fixed

File: tests/copy.bats (hunk 4 above)
Comment (rhatdan) [inline tests/copy.bats]: Fixed

File: tests/copy.bats (hunk 5 above)
Comment (rhatdan) [inline tests/copy.bats]: Fixed

**Comment by rhatdan** [comment]
```
Multiple ArgsUsage field values in main.go also need updates.
```
Can you give me an example of one that is wrong?

File: cmd/buildah/commit.go [hunk 6]
```diff
@@ -11,24 +11,11 @@ import (
...
-			Name:   "root",
-			Usage:  "root `directory` of the working container",
-			EnvVar: "BUILDAHROOT",
-		},
 		cli.BoolFlag{
 			Name:  "disable-compression",
 			Usage: "don't compress layers",
 		},
 		cli.StringFlag{
-			Name:  "output",
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/commit.go]: just double checking that the output param should be out nwo.

File: cmd/buildah/commit.go [hunk 7]
```diff
@@ -38,16 +25,17 @@ var (
...
 	}
-	root := c.String("root")
-
-	output := ""
-	if c.IsSet("output") {
-		output = c.String("output")
+	name := args[0]
+	args = args.Tail()
+	if len(args) == 0 {
+		return fmt.Errorf("an image name must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/commit.go]: I don't think we can ever get into this if statement.  Plus we seem to be setting the name and image to the same value?  Is that correct?

File: cmd/buildah/commit.go (hunk 6 above)
Comment (rhatdan) [inline cmd/buildah/commit.go]: Yes I fixed up commit also.  I guess I need to update the pull request.

Comment (nalind) [inline cmd/buildah/commit.go]: Yes, it should be; it's supplied in the second argument now.

File: cmd/buildah/commit.go (hunk 7 above)
Comment (nalind) [inline cmd/buildah/commit.go]: The Tail() call immediately before essentially sets args to args[1:], shifting everything one subscript lower in the slice.

**Comment by nalind** [comment]
@rhatdan My mistake - we already updated those in a different PR.

File: cmd/buildah/delete.go [hunk 8]
```diff
@@ -7,41 +7,23 @@ import (
...
-		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
-		}
-		name = args[0]
-		args = args.Tail()
+	if len(args) == 0 {
+		return fmt.Errorf("container ID must be specified")
 	}
+	name := args[0]
+	args = args.Tail()
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/delete.go]: I don't think this args = args.Tail() line is necessary as I'm not seeing a reference to args after this.

File: cmd/buildah/commit.go (hunk 7 above)
Comment (rhatdan) [inline cmd/buildah/commit.go]: No first arg[0] should be set to name second to image.

File: tests/basic.bats [hunk 9]
```diff
@@ -3,93 +3,93 @@
...
+  cid=$(buildah from --signature-policy ${TESTSDIR}/policy.json --image alpine)
+  buildah delete        $cid
 }
 
 @test "from-nopull" {
-	run buildah from --pull=false --signature-policy ${TESTSDIR}/policy.json --image alpine
-	[ "$status" -eq 1 ]
+  run buildah from --pull=false --signature-policy ${TESTSDIR}/policy.json --image alpine
+  [ "$status" -eq 1 ]
 }
```
Comment (nalind) [inline tests/basic.bats]: What's up with all the reindenting?

File: cmd/buildah/delete.go (hunk 8 above)
Comment (rhatdan) [inline cmd/buildah/delete.go]: Maybe we should check if args still has data and fail

File: tests/basic.bats (hunk 9 above)
Comment (rhatdan) [inline tests/basic.bats]: Whitespace cleanup in emacs did this.

Comment (rhatdan) [inline tests/basic.bats]: Do you want these 8 spaces?

File: cmd/buildah/umount.go [hunk 10]
```diff
@@ -7,41 +7,23 @@ import (
...
-		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
-		}
-		name = args[0]
-		args = args.Tail()
+	if len(args) == 0 {
+		return fmt.Errorf("container ID must be specified")
 	}
+	name := args[0]
+	args = args.Tail()
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/umount.go]: Ditto
Not seeing args used after this point.

File: tests/add.bats [hunk 11]
```diff
@@ -3,95 +3,95 @@
...
+  cid=$(buildah from --pull --signature-policy ${TESTSDIR}/policy.json --image alpine)
+  root=$(buildah mount $cid)
+  mkdir $root/subdir $root/other-subdir
+  # Copy a file to the working directory
+  buildah config --workingdir=/ $cid
+  buildah add $cid ${TESTDIR}/randomfile
+  # Copy a file to a specific subdirectory
+  buildah add --dest=/subdir $cid ${TESTDIR}/randomfile
+  # Copy a file two files to a specific subdirectory
+  buildah add --dest=/other-subdir  $cid ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile
```
Comment (TomSweeneyRedHat) [inline tests/add.bats]: double space before $cid

File: tests/add.bats [hunk 12]
```diff
@@ -3,95 +3,95 @@
...
+  # Copy a file two files to a specific location, which fails because it's not a directory.
+  run buildah add --dest=/notthereyet-subdir ${TESTDIR}/randomfile ${TESTDIR}/other-randomfile  $cid
```
Comment (TomSweeneyRedHat) [inline tests/add.bats]: double space before $cid

File: tests/basic.bats [hunk 13]
```diff
@@ -3,93 +3,93 @@
...
-	buildah unmount --name=$newcid
-	buildah delete --name=$newcid
+  newcid=$(buildah from --image new-image)
+  newroot=$(buildah mount $newcid)
+  test -s $newroot/randomfile
+  cmp ${TESTDIR}/randomfile $newroot/randomfile
+  cp ${TESTDIR}/other-randomfile $newroot/other-randomfile
+  buildah commit --signature-policy ${TESTSDIR}/policy.json $newcid containers-storage:other-new-image
+  # Not an allowed ordering of arguments and flags.  Check that it's rejected.
+  run buildah commit $newcid  --signature-policy ${TESTSDIR}/policy.json containers-storage:rejected-new-image
```
Comment (TomSweeneyRedHat) [inline tests/basic.bats]: double space after $newcid

File: cmd/buildah/umount.go (hunk 10 above)
Comment (rhatdan) [inline cmd/buildah/umount.go]: Now checks for too many arguments.

File: tests/add.bats [hunk 14]
```diff
@@ -3,95 +3,95 @@
...
-	buildah delete --name=$cid
+  cid=$(buildah from --pull --signature-policy ${TESTSDIR}/policy.json --image alpine)
+  root=$(buildah mount $cid)
+  mkdir $root/subdir $root/other-subdir
+  # Copy a file to the working directory
+  buildah config --workingdir=/ $cid
+  buildah add $cid ${TESTDIR}/randomfile
+  # Copy a file to a specific subdirectory
+  buildah add --dest=/subdir $cid ${TESTDIR}/randomfile
+  # Copy a file two files to a specific subdirectory
```
Comment (nalind) [inline tests/add.bats]: Might as well fix this comment while we're in here, then.

File: tests/basic.bats (hunk 9 above)
Comment (nalind) [inline tests/basic.bats]: I'm not particularly attached to them, but if the patch isn't already making non-whitespace changes to the same lines, it just makes the patch longer and a bit harder to review.

File: tests/add.bats (hunk 14 above)
Comment (rhatdan) [inline tests/add.bats]: Fixed

**Comment by rhatdan** [comment]
I think I have fixed all comments except for the whitespace change.  Can't figure out how to make emacs change the tab size.

**Comment by nalind** [comment]
[[Buildah] (8) buildah CLI overhaul](https://trello.com/c/LULizZlk/442-buildah-8-buildah-cli-overhaul)

**Pull request merged**